### ISSUE TITLE
storage location of dynamic seds 

### ISSUE THREAD

**Issue by rbiswas4** [LSSTDESC/sims_GCRCatSimInterface#21]
Currently, the sprinkler and the sn codes write out seds to files  in a directory called `spectra_files/Dynamic` relative to the working directory from which the script is launched. Each row in the phosim instance catalog corresponds to such a SN and lists the filename relative to the `spectra_files/Dynamic` directory. An example of how the sed filename  looks in the instance catalog row is `specFileSN_3322_59580.1396_r.dat` 

@TomGlanzman and I discussed this and agreed we would have the following changes that will help the production code and storage:
- the seds will be written out to an `out_dir` specified by a production script. ie. `spectra_files/Dynamic` is replaced by `args.out_dir/Dynamic`
- The filename in the row of the phosim instance catalog should read `Dynamic/filename`. so in the example above `Dynamic/specFileSN_3322_59580.1396_r.dat` 

@TomGlanzman would like to see this in a test on Cori, probably repeating @jchiang87 's script. To set this up, we would need to have 
- the global lsst stack on cori
- masters of the following lsst repos and Twinkles (along with its dependencies `OM10` and `lenspop` that could be setup using the following script:
```
source ${lsst_stack}
export DC2_inst= < location of these repos >
setup lsst_sims
setup -r ${DC2_inst}/sims_GCRCatSimInterface -j
setup -r ${DC2_inst}/sims_catUtils -j
setup -r ${DC2_inst}/sims_utils -j
setup -r ${DC2_inst}/Twinkles -j
setup -r ${DC2_inst}/imSim -j
setup -r ${DC2_inst}/sims_GalSimInterface -j 

mkdir -p /global/cscratch1/sd/rbiswas/DC2_test_files
cp /global/cscratch1/sd/jchiang8/DC2_prs/agn_db_mbh_7.0_m_i_30.0.sqlite $SCRATCH/DC2_test_files/
cp /global/cscratch1/sd/jchiang8/DC2_prs/agn_db_mbh_7.0_m_i_30.0.sqlite $SCRATCH/DC2_test_files/
```
The OM10 repo can be installed on nersc using the instructions from the OM10 repository:
```
#python setup.py develop
#Now when you import om10 you'll get the development version, linked to to your fork.

Make sure you have all the requirements with

#conda install scipy>=0.18.1
#pip install -r requirements.txt```

checkout the branch corresponding to this issue in `sims_GCRCatSimInterface` and in  the `bin.src` run 
```
#### Please note @jchiang8 would prefer us to not install om10  and lenspop this way and use the `pythonpath`. will update methods once I work it out. Also note slight change in running of test.
```

```
nohup ./generateInstCat.py --db /global/projecta/projectdirs/lsst/groups/SSim/DC2/minion_1016_desc_dithered_v4.db --agn_db_name $SCRATCH/DC2_test_files/agn_db_mbh_7.0_m_i_30.0.sqlite --descqa_catalog protoDC2 --out_dir sprinkled_sn_sl_0.5 --ids 230 --fov 0.5 --protoDC2_ra 55.064 --protoDC2_dec -29.783 --enable_sprinkler > large_test.log 2>&1 &
```

**Comment by jchiang87** [comment]
Don't forget about the SED files written for the lensed SNe! See Bryce's comment at https://github.com/LSSTDESC/sims_GCRCatSimInterface/pull/19#issuecomment-362346017 and the subsequent comment by me.

**Comment by TomGlanzman** [comment]
@rbiswas4 With regard to your earlier post, here is the LSST stack setup I am currently using for DC2-phoSim.

 /global/common/software/lsst/cori-haswell-gcc/stack/setup_w_2017_46_py3_gcc6.sh

**Comment by TomGlanzman** [comment]
I've looked at @rbiswas4 's latest test and it looks good!  A new "Dynamic" subdirectory was created in the instanceCatalog out_dir.  The catalog objects point to, e.g., Dynamic/specFileSN_3322_59580.1396_r.dat
PhoSim can be coerced to look there...:)

**Comment by rbiswas4** [comment]
Closed via #22
